src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -0,0 +1,174 @@
+     * @return an iterable of size >= 1. if size==1, the returned iterable contains only the input (i.e. either no gap or hasn't reached sensitivity)
+     */
+    @VisibleForTesting
+    static Iterable<AlignmentRegion> breakGappedAlignment(final AlignmentRegion oneRegion, final int sensitivity) {
+
+        final List<CigarElement> cigarElements = oneRegion.forwardStrandCigar.getCigarElements();
+        if(cigarElements.size()==1) return new ArrayList<>( Collections.singletonList(oneRegion) );
+        final int gapCount = (int) cigarElements.stream().filter(ce -> ce.getOperator().isIndel() && ce.getLength() >= sensitivity).count();
+        if (gapCount==0) return new ArrayList<>( Collections.singletonList(oneRegion) );
+
+        // get necessary information for computing
+        final String asmId = oneRegion.assemblyId;
+        final String contigId = oneRegion.contigId;
+        final String chr = oneRegion.referenceInterval.getContig();
+        final Cigar originalForwardStrandCigar = oneRegion.forwardStrandCigar;
+        final boolean isForwardStrand = oneRegion.forwardStrand;
+        final int newMapQual = oneRegion.mapQual;
+        final int refStart = oneRegion.referenceInterval.getStart();
+        final int refEnd   = oneRegion.referenceInterval.getEnd();
+        final int contigLength = oneRegion.assembledContigLength;
+
+        // needs to accomplish three key functionality: 1. generate appropriate cigar, 2. infer reference coordinates, 3. infer contig coordinates, (cannot do yet: 4. mismatches and mapping quality)
+        final List<AlignmentRegion> result = new ArrayList<>(gapCount+1);
+
+        final List<CigarElement> cigarMemoryList = new ArrayList<>();
+        final int clippedNBases = SVVariantCallerUtils.getNumClippedBases(true, originalForwardStrandCigar);
+        final int hardClippingAtEnd = (cigarElements.get(cigarElements.size()-1).getOperator()== CigarOperator.H)? cigarElements.get(cigarElements.size()-1).getLength() : 0;
+        int refIntervalStart = refStart, contigIntervalStart = 1 + clippedNBases;
+        int gapIndex=0;
+        for(int i=0; i<cigarElements.size(); ++i){ // considers only 'M', 'I', 'D', 'S', and 'H'
+            final CigarElement current = cigarElements.get(i);
+            final CigarOperator op = current.getOperator();
+            final int operatorLen = current.getLength();
+            switch (op){
+                case M: case S: case H:
+                    cigarMemoryList.add(current);
+                    break;
+                case I: case D:
+                    if (operatorLen<sensitivity) {
+                        cigarMemoryList.add(current);
+                        break;
+                    }
+
+                    // collapse cigar memory list into a single cigar
+                    final List<CigarElement> cigarMemoryListModifiable = new ArrayList<>(cigarMemoryList);
```

Comment:
The intension was to make a local variable (the `cigarMemoryListModifiable`) to compute what CIGAR should be output for the newly generated alignments, i.e. increase readability.
In terms of performance, the original `cigarMemoryList` is going to be cleared anyway, so from that perspective it makes sense not to create the copy.
Am I understanding the issue correctly?